cmd/image-builder/config.go

```diff
@@ -102,3 +114,161 @@ func (r *Registry) UnmarshalYAML(value *yaml.Node) error {
 	*r = regs
 	return nil
 }
+
+// GitStateConfig holds information about repository and specific commit
+// from which image should be build.
+// It also contains information whether job is presubmit or postsubmit
+type GitStateConfig struct {
+	// Name of the source repository
+	RepositoryName string
+	// Name of the source repository's owner
+	RepositoryOwner string
+	// Type of the job, allowed values "presubmit" or "postsubmit"
+	JobType string
+	// Number of the pull request for presubmit job
+	PullRequestNumber int
+	// Commit SHA for base branch
+	BaseCommitSHA string
+	// Commit SHA for head of the pull request
+	PullHeadCommitSHA string
+	// isPullRequest contains information whether event which triggered the job was from pull request
+	isPullRequest bool
+}
+
+func (gitState GitStateConfig) IsPullRequest() bool {
+	return gitState.isPullRequest
+}
+
+func LoadGitStateConfigFromEnv(o options) (GitStateConfig, error) {
+	switch o.ciSystem {
+	// Load from env specific for github actions
+	case GithubActions:
+		return loadGithubActionsGitState()
+	// Load from env specific for prow jobs
+	case Prow:
+		return loadProwJobGitState()
+	default:
+		// Unknown CI System, return error and empty git state
+		return GitStateConfig{}, fmt.Errorf("unknown ci system, got %s", o.ciSystem)
+	}
+}
+
+func loadProwJobGitState() (GitStateConfig, error) {
+	repoName, present := os.LookupEnv("REPO_NAME")
+	if !present {
+		return GitStateConfig{}, fmt.Errorf("REPO_NAME environment variable is not set, please set it to valid repository name")
+	}
+
+	repoOwner, present := os.LookupEnv("REPO_OWNER")
+	if !present {
+		return GitStateConfig{}, fmt.Errorf("REPO_OWNER environment variable is not set, please set it to valid repository owner")
+	}
+
+	jobType, present := os.LookupEnv("JOB_TYPE")
+	if !present {
+		return GitStateConfig{}, fmt.Errorf("JOB_TYPE environment variable is not set, please set it to valid job type")
+	}
+	if !slices.Contains([]string{"presubmit", "postsubmit"}, jobType) {
+		return GitStateConfig{}, fmt.Errorf("JOB_TYPE environment variable is not set to valid value, please set it to either 'presubmit' or 'postsubmit'")
```

Comment:
I would rather write an error message that image-builder is running for unsupported event.